src/ctap2/attestation.rs

```diff
@@ -220,10 +220,7 @@ fn parse_ad<'a>(input: &'a [u8]) -> IResult<&'a [u8], AuthenticatorData, NomErro
     // AuthenticatorDataFlags, just truncate the one we don't know
     let (rest, flags) = map(be_u8, AuthenticatorDataFlags::from_bits_truncate)(rest)?;
     let (rest, counter) = be_u32(rest)?;
-    let (rest, credential_data) = cond(
-        flags.contains(AuthenticatorDataFlags::ATTESTED),
-        parse_attested_cred_data,
-    )(rest)?;
+    let (rest, credential_data) = opt(parse_attested_cred_data)(rest)?;
```

Comment:
OK, but I think we should enforce the requirement that `credential_data` be present if and only if the AT flag is set.